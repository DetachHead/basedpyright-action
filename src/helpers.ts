import * as core from '@actions/core';
import * as httpClient from '@actions/http-client';
import * as tc from '@actions/tool-cache';
import * as path from 'path';
import SemVer from 'semver/classes/semver';
import stringArgv from 'string-argv';

import { NpmRegistryResponse } from './schema';

export function getNodeInfo() {
    return {
        version: process.version,
        execPath: process.execPath,
    };
}

export async function getArgs() {
    const pyrightVersion = await getPyrightVersion();
    const pyrightPath = await downloadPyright(pyrightVersion);

    const args = [pyrightPath];

    const workingDirectory = core.getInput('working-directory');

    const noComments = getBooleanInput('no-comments', false);
    if (!noComments) {
        args.push('--outputjson');
    }

    const pythonPlatform = core.getInput('python-platform');
    if (pythonPlatform) {
        args.push('--pythonplatform');
        args.push(pythonPlatform);
    }

    const pythonVersion = core.getInput('python-version');
    if (pythonVersion) {
        args.push('--pythonversion');
        args.push(pythonVersion);
    }

    const typeshedPath = core.getInput('typeshed-path');
    if (typeshedPath) {
        args.push('--typeshed-path');
        args.push(typeshedPath);
    }

    const venvPath = core.getInput('venv-path');
    if (venvPath) {
        args.push('--venv-path');
        args.push(venvPath);
    }

    const project = core.getInput('project');
    if (project) {
        args.push('--project');
        args.push(project);
    }

    const lib = getBooleanInput('lib', false);
    if (lib) {
        args.push('--lib');
    }

    const warnings = getBooleanInput('warnings', false);
    if (warnings) {
        args.push('--warnings');
    }

    const verifyTypes = core.getInput('verify-types');
    if (verifyTypes) {
        args.push('--verifytypes');
        args.push(verifyTypes);
    }

    const extraArgs = core.getInput('extra-args');
    if (extraArgs) {
        args.push(...stringArgv(extraArgs));
    }

    return {
        workingDirectory,
        noComments,
        pyrightVersion,
        args,
    };
}

function getBooleanInput(name: string, defaultValue: boolean): boolean {
    const input = core.getInput(name);
    if (!input) {
        return defaultValue;
    }
    return input.toUpperCase() === 'TRUE';
}

async function getPyrightVersion(): Promise<SemVer> {
    const versionSpec = core.getInput('version');
    if (versionSpec) {
        return new SemVer(versionSpec);
    }

    const client = new httpClient.HttpClient();
    const resp = await client.get('https://registry.npmjs.org/pyright/latest');
    const body = await resp.readBody();
    const obj = NpmRegistryResponse.parse(JSON.parse(body));
    return new SemVer(obj.version);
}

async function downloadPyright(version: SemVer): Promise<string> {
    // Note: this only works because the pyright package doesn't have any
    // dependencies. If this ever changes, we'll have to actually install it.
    const url = `https://registry.npmjs.org/pyright/-/pyright-${version.format()}.tgz`;
    const pyrightTarball = await tc.downloadTool(url);
    const pyright = await tc.extractTar(pyrightTarball);
    return path.join(pyright, 'package', 'index.js');
}
