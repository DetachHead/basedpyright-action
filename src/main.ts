import * as core from '@actions/core';
import * as command from '@actions/core/lib/command';
import * as tc from '@actions/tool-cache';
import chalk from 'chalk';
import * as cp from 'child_process';
import * as path from 'path';
import SemVer from 'semver/classes/semver';
import stringArgv from 'string-argv';

import { Diagnostic, isEmptyRange, Report } from './schema';

export async function main() {
    try {
        const cwd = core.getInput('working-directory');
        if (cwd) {
            process.chdir(cwd);
        }

        const args = await getArgs();

        const { status, stdout } = cp.spawnSync(process.execPath, args, {
            encoding: 'utf-8',
            stdio: ['ignore', 'pipe', 'inherit'],
        });

        if (!stdout.trim()) {
            // Process crashed. stderr was inherited, so just mark the step as failed.
            core.setFailed(`Exit code ${status}`);
            return;
        }

        const report = Report.parse(JSON.parse(stdout));

        report.generalDiagnostics.forEach((diag) => {
            // TODO: Only do this logging on info?
            logDiagnosticToConsole(diag);

            if (diag.severity === 'information') {
                return;
            }

            const line = diag.range?.start.line ?? 0;
            const col = diag.range?.start.character ?? 0;
            const message = diag.rule ? `${diag.rule}: ${diag.message}` : diag.message; // TODO: Newlines make GHA's issue reporting unhappy.

            command.issueCommand(
                diag.severity,
                {
                    file: diag.file,
                    line: line + 1,
                    col: col + 1,
                },
                message
            );
        });

        const { errorCount, warningCount, informationCount } = report.summary;

        console.log(
            `${errorCount} ${errorCount === 1 ? 'error' : 'errors'}, ` +
                `${warningCount} ${warningCount === 1 ? 'warning' : 'warnings'}, ` +
                `${informationCount} ${informationCount === 1 ? 'info' : 'infos'} `
        );

        if (status !== 0) {
            core.setFailed(`${errorCount} ${errorCount === 1 ? 'error' : 'errors'}`);
        }
    } catch (e) {
        core.setFailed(e.message);
    }
}

async function getArgs(): Promise<string[]> {
    const versionSpec = core.getInput('version');
    const version = versionSpec ? new SemVer(versionSpec) : undefined;

    const pyrightIndex = await getPyright(version);

    const args = [pyrightIndex, '--outputjson'];

    const pythonPlatform = core.getInput('python-platform');
    if (pythonPlatform) {
        args.push('--pythonplatform');
        args.push(pythonPlatform);
    }

    const pythonVersion = core.getInput('python-version');
    if (pythonVersion) {
        args.push('--pythonversion');
        args.push(pythonVersion);
    }

    const typeshedPath = core.getInput('typeshed-path');
    if (typeshedPath) {
        args.push('--typeshed-path');
        args.push(typeshedPath);
    }

    const venvPath = core.getInput('venv-path');
    if (venvPath) {
        args.push('--venv-path');
        args.push(venvPath);
    }

    const project = core.getInput('project');
    if (project) {
        args.push('--project');
        args.push(project);
    }

    const lib = (core.getInput('lib') || 'false').toUpperCase() === 'TRUE';
    if (lib) {
        args.push('--lib');
    }

    const extraArgs = core.getInput('extra-args');
    if (extraArgs) {
        args.push(...stringArgv(extraArgs));
    }

    return args;
}

function pyrightUrl(version?: SemVer): string {
    if (!version) {
        // TODO: Do this better.
        return cp.execFileSync('npm', ['view', 'pyright', 'dist.tarball'], { encoding: 'utf-8' }).trim();
    }
    return `https://registry.npmjs.org/pyright/-/pyright-${version.format()}.tgz`;
}

async function getPyright(version?: SemVer): Promise<string> {
    const url = pyrightUrl(version);

    const pyrightTarball = await tc.downloadTool(url);
    const pyright = await tc.extractTar(pyrightTarball);
    return path.join(pyright, 'package', 'index.js');
}

// Copied from pyright.
function logDiagnosticToConsole(diag: Diagnostic, prefix = '') {
    let message = prefix;
    if (diag.file) {
        message += `${diag.file}:`;
    }
    if (diag.range && !isEmptyRange(diag.range)) {
        message +=
            chalk.yellow(`${diag.range.start.line + 1}`) +
            ':' +
            chalk.yellow(`${diag.range.start.character + 1}`) +
            ' - ';
    }

    const [firstLine, ...remainingLines] = diag.message.split('\n');

    message +=
        diag.severity === 'error'
            ? chalk.red('error')
            : diag.severity === 'warning'
            ? chalk.cyan('warning')
            : chalk.blue('info');
    message += `: ${firstLine}`;
    if (remainingLines.length > 0) {
        message += '\n' + prefix + remainingLines.join('\n' + prefix);
    }

    if (diag.rule) {
        message += chalk.gray(` (${diag.rule})`);
    }

    console.log(message);
}
