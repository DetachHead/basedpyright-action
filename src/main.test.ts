/* eslint-disable @typescript-eslint/no-unsafe-assignment */
/* eslint-disable unicorn/no-null */
import * as cp from "node:child_process";

import * as core from "@actions/core";
import * as command from "@actions/core/lib/command";
import { afterEach, beforeEach, describe, expect, test, vitest } from "vitest";

import * as helpers from "./helpers";

const nodeVersion = "v20.8.1";
const nodeExecPath = "/path/to/node";
const pyrightVersion = "1.1.240";

vitest.mock("@actions/core");
const mockedCore = vitest.mocked(core);
vitest.mock("@actions/core/lib/command");
const mockedCommand = vitest.mocked(command);
vitest.mock("node:child_process");
const mockedCp = vitest.mocked(cp);
vitest.mock("./helpers");
const mockedHelpers = vitest.mocked(helpers);

const mockedProcessChdir = vitest.spyOn(process, "chdir");
const mockedProcessCwd = vitest.spyOn(process, "cwd");

let currentWorkingDirectory = "/some/default/cwd";

import { main } from "./main";
import type { Report } from "./schema";

beforeEach(() => {
    vitest.clearAllMocks();
    mockedHelpers.getNodeInfo.mockReturnValue({
        version: nodeVersion,
        execPath: nodeExecPath,
    });
    mockedHelpers.getActionVersion.mockReturnValue("1.1.0");
    mockedProcessChdir.mockImplementation((dir) => {
        currentWorkingDirectory = dir;
    });
    mockedProcessCwd.mockImplementation(() => currentWorkingDirectory);
});

afterEach(() => {
    expect(mockedProcessChdir.mock.calls).toMatchSnapshot("process.chdir");
    expect(mockedCore.setFailed.mock.calls).toMatchSnapshot("core.setFailed");
    expect(mockedCore.info.mock.calls).toMatchSnapshot("core.info");
    expect(mockedCommand.issueCommand.mock.calls).toMatchSnapshot("command.issueCommand");
    expect(mockedCp.spawnSync.mock.calls).toMatchSnapshot("cp.spawnSync");
});

test("thrown error at first call", async () => {
    mockedHelpers.getNodeInfo.mockImplementation(() => {
        throw new Error("oops");
    });

    await main();
});

describe("no comments", () => {
    const args = ["/path/to/pyright/dist/index.js", "--outputjson"];
    const wd = "/some/wd";

    beforeEach(() => {
        mockedHelpers.getArgs.mockResolvedValue({
            noComments: true,
            workingDirectory: wd,
            pyrightVersion,
            args,
        });
    });

    test("success", async () => {
        mockedCp.spawnSync.mockImplementation(() => ({
            pid: -1,
            output: [],
            stdout: "" as any,
            stderr: "" as any,
            status: 0,
            signal: null,
        }));

        await main();
    });

    test("failure", async () => {
        mockedCp.spawnSync.mockImplementation(() => ({
            pid: -1,
            output: [],
            stdout: "" as any,
            stderr: "" as any,
            status: 1,
            signal: null,
        }));

        await main();
    });
});

describe("with comments", () => {
    const args = ["/path/to/pyright/dist/index.js"];

    beforeEach(() => {
        mockedHelpers.getArgs.mockResolvedValue({
            noComments: false,
            workingDirectory: "",
            pyrightVersion,
            args,
        });
    });

    test("failure", async () => {
        mockedCp.spawnSync.mockImplementation(() => ({
            pid: -1,
            output: [],
            stdout: "" as any,
            stderr: "" as any,
            status: 2,
            signal: null,
        }));

        await main();
    });

    test("unparsable json", async () => {
        mockedCp.spawnSync.mockImplementation(() => ({
            pid: -1,
            output: [],
            stdout: "}" as any,
            stderr: "" as any,
            status: 0,
            signal: null,
        }));

        await main();
    });

    test("invalid stdout", async () => {
        mockedCp.spawnSync.mockImplementation(() => ({
            pid: -1,
            output: [],
            stdout: "{}" as any,
            stderr: "" as any,
            status: 0,
            signal: null,
        }));

        await main();
    });

    test("no diagnostics", async () => {
        mockedCp.spawnSync.mockImplementation(() => ({
            pid: -1,
            output: [],
            stdout: reportToString({
                generalDiagnostics: [],
                summary: {
                    errorCount: 0,
                    warningCount: 0,
                    informationCount: 0,
                },
            }) as any,
            stderr: "" as any,
            status: 0,
            signal: null,
        }));

        await main();
    });

    test("with diagnostics", async () => {
        mockedCp.spawnSync.mockImplementation(() => ({
            pid: -1,
            output: [],
            stdout: reportToString({
                generalDiagnostics: [
                    {
                        file: "/path/to/file1.py",
                        range: { start: { line: 0, character: 0 }, end: { line: 1, character: 1 } },
                        severity: "error",
                        message: "some error",
                    },
                    {
                        file: "/path/to/file2.py",
                        range: { start: { line: 0, character: 0 }, end: { line: 0, character: 0 } },
                        severity: "warning",
                        message: "some warning",
                    },
                    {
                        file: "/path/to/file3.py",
                        range: { start: { line: 0, character: 0 }, end: { line: 1, character: 1 } },
                        severity: "information",
                        message: "some info",
                        rule: "reportSomeInformation",
                    },
                    {
                        file: "/path/to/file3.py",
                        severity: "warning",
                        message: "another warning",
                        rule: "reportSomeWarning",
                    },
                    {
                        file: "/path/to/file1.py",
                        range: { start: { line: 5, character: 9 }, end: { line: 5, character: 15 } },
                        severity: "error",
                        message: "some error",
                    },
                ],
                summary: {
                    errorCount: 2,
                    warningCount: 2,
                    informationCount: 1,
                },
            }) as any,
            stderr: "" as any,
            status: 1,
            signal: null,
        }));

        await main();
    });

    test("with 1 each", async () => {
        mockedCp.spawnSync.mockImplementation(() => ({
            pid: -1,
            output: [],
            stdout: reportToString({
                generalDiagnostics: [
                    {
                        file: "/path/to/file1.py",
                        range: { start: { line: 0, character: 0 }, end: { line: 1, character: 1 } },
                        severity: "error",
                        message: "some error",
                    },
                    {
                        file: "/path/to/file2.py",
                        range: { start: { line: 0, character: 0 }, end: { line: 0, character: 0 } },
                        severity: "warning",
                        message: "some warning",
                    },
                    {
                        file: "/path/to/file3.py",
                        range: { start: { line: 0, character: 0 }, end: { line: 1, character: 1 } },
                        severity: "information",
                        message: "some info",
                        rule: "reportSomeInformation",
                    },
                ],
                summary: {
                    errorCount: 1,
                    warningCount: 1,
                    informationCount: 1,
                },
            }) as any,
            stderr: "" as any,
            status: 1,
            signal: null,
        }));

        await main();
    });
});

function reportToString(report: Report): string {
    return JSON.stringify(report);
}
