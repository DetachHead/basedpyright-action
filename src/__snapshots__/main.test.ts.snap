// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`no comments > failure > command.issueCommand 1`] = `[]`;

exports[`no comments > failure > core.info 1`] = `
[
  [
    "pyright 1.1.240, node v20.8.1, pyright-action 1.1.0",
  ],
  [
    "Working directory: /some/wd",
  ],
  [
    "Running: /path/to/node /path/to/pyright/dist/index.js --outputjson",
  ],
]
`;

exports[`no comments > failure > core.setFailed 1`] = `
[
  [
    "Exit code 1",
  ],
]
`;

exports[`no comments > failure > cp.spawnSync 1`] = `
[
  [
    "/path/to/node",
    [
      "/path/to/pyright/dist/index.js",
      "--outputjson",
    ],
    {
      "stdio": [
        "ignore",
        "inherit",
        "inherit",
      ],
    },
  ],
]
`;

exports[`no comments > failure > process.chdir 1`] = `
[
  [
    "/some/wd",
  ],
]
`;

exports[`no comments > success > command.issueCommand 1`] = `[]`;

exports[`no comments > success > core.info 1`] = `
[
  [
    "pyright 1.1.240, node v20.8.1, pyright-action 1.1.0",
  ],
  [
    "Working directory: /some/wd",
  ],
  [
    "Running: /path/to/node /path/to/pyright/dist/index.js --outputjson",
  ],
]
`;

exports[`no comments > success > core.setFailed 1`] = `[]`;

exports[`no comments > success > cp.spawnSync 1`] = `
[
  [
    "/path/to/node",
    [
      "/path/to/pyright/dist/index.js",
      "--outputjson",
    ],
    {
      "stdio": [
        "ignore",
        "inherit",
        "inherit",
      ],
    },
  ],
]
`;

exports[`no comments > success > process.chdir 1`] = `
[
  [
    "/some/wd",
  ],
]
`;

exports[`thrown error at first call > command.issueCommand 1`] = `[]`;

exports[`thrown error at first call > core.info 1`] = `[]`;

exports[`thrown error at first call > core.setFailed 1`] = `
[
  [
    [Error: oops],
  ],
]
`;

exports[`thrown error at first call > cp.spawnSync 1`] = `[]`;

exports[`thrown error at first call > process.chdir 1`] = `[]`;

exports[`with comments > failure > command.issueCommand 1`] = `[]`;

exports[`with comments > failure > core.info 1`] = `
[
  [
    "pyright 1.1.240, node v20.8.1, pyright-action 1.1.0",
  ],
  [
    "Working directory: /some/wd",
  ],
  [
    "Running: /path/to/node /path/to/pyright/dist/index.js --outputjson",
  ],
]
`;

exports[`with comments > failure > core.setFailed 1`] = `
[
  [
    "Exit code 2",
  ],
]
`;

exports[`with comments > failure > cp.spawnSync 1`] = `
[
  [
    "/path/to/node",
    [
      "/path/to/pyright/dist/index.js",
      "--outputjson",
    ],
    {
      "encoding": "utf8",
      "maxBuffer": 104857600,
      "stdio": [
        "ignore",
        "pipe",
        "inherit",
      ],
    },
  ],
]
`;

exports[`with comments > failure > process.chdir 1`] = `[]`;

exports[`with comments > invalid stdout > command.issueCommand 1`] = `[]`;

exports[`with comments > invalid stdout > core.info 1`] = `
[
  [
    "pyright 1.1.240, node v20.8.1, pyright-action 1.1.0",
  ],
  [
    "Working directory: /some/wd",
  ],
  [
    "Running: /path/to/node /path/to/pyright/dist/index.js --outputjson",
  ],
]
`;

exports[`with comments > invalid stdout > core.setFailed 1`] = `
[
  [
    [ValitaError: missing_value at .generalDiagnostics (missing value) (+ 1 other issue)],
  ],
]
`;

exports[`with comments > invalid stdout > cp.spawnSync 1`] = `
[
  [
    "/path/to/node",
    [
      "/path/to/pyright/dist/index.js",
      "--outputjson",
    ],
    {
      "encoding": "utf8",
      "maxBuffer": 104857600,
      "stdio": [
        "ignore",
        "pipe",
        "inherit",
      ],
    },
  ],
]
`;

exports[`with comments > invalid stdout > process.chdir 1`] = `[]`;

exports[`with comments > no diagnostics > command.issueCommand 1`] = `[]`;

exports[`with comments > no diagnostics > core.info 1`] = `
[
  [
    "pyright 1.1.240, node v20.8.1, pyright-action 1.1.0",
  ],
  [
    "Working directory: /some/wd",
  ],
  [
    "Running: /path/to/node /path/to/pyright/dist/index.js --outputjson",
  ],
  [
    "0 errors, 0 warnings, 0 informations",
  ],
]
`;

exports[`with comments > no diagnostics > core.setFailed 1`] = `[]`;

exports[`with comments > no diagnostics > cp.spawnSync 1`] = `
[
  [
    "/path/to/node",
    [
      "/path/to/pyright/dist/index.js",
      "--outputjson",
    ],
    {
      "encoding": "utf8",
      "maxBuffer": 104857600,
      "stdio": [
        "ignore",
        "pipe",
        "inherit",
      ],
    },
  ],
]
`;

exports[`with comments > no diagnostics > process.chdir 1`] = `[]`;

exports[`with comments > unparsable json > command.issueCommand 1`] = `[]`;

exports[`with comments > unparsable json > core.info 1`] = `
[
  [
    "pyright 1.1.240, node v20.8.1, pyright-action 1.1.0",
  ],
  [
    "Working directory: /some/wd",
  ],
  [
    "Running: /path/to/node /path/to/pyright/dist/index.js --outputjson",
  ],
]
`;

exports[`with comments > unparsable json > core.setFailed 1`] = `
[
  [
    [SyntaxError: Unexpected token '}', "}" is not valid JSON],
  ],
]
`;

exports[`with comments > unparsable json > cp.spawnSync 1`] = `
[
  [
    "/path/to/node",
    [
      "/path/to/pyright/dist/index.js",
      "--outputjson",
    ],
    {
      "encoding": "utf8",
      "maxBuffer": 104857600,
      "stdio": [
        "ignore",
        "pipe",
        "inherit",
      ],
    },
  ],
]
`;

exports[`with comments > unparsable json > process.chdir 1`] = `[]`;

exports[`with comments > with 1 each > command.issueCommand 1`] = `
[
  [
    "error",
    {
      "col": 1,
      "file": "/path/to/file1.py",
      "line": 1,
    },
    "some error",
  ],
  [
    "warning",
    {
      "col": 1,
      "file": "/path/to/file2.py",
      "line": 1,
    },
    "some warning",
  ],
]
`;

exports[`with comments > with 1 each > core.info 1`] = `
[
  [
    "pyright 1.1.240, node v20.8.1, pyright-action 1.1.0",
  ],
  [
    "Working directory: /some/wd",
  ],
  [
    "Running: /path/to/node /path/to/pyright/dist/index.js --outputjson",
  ],
  [
    "/path/to/file1.py:1:1 - error: some error",
  ],
  [
    "/path/to/file2.py: warning: some warning",
  ],
  [
    "/path/to/file3.py:1:1 - information: some info (reportSomeInformation)",
  ],
  [
    "1 error, 1 warning, 1 information",
  ],
]
`;

exports[`with comments > with 1 each > core.setFailed 1`] = `
[
  [
    "1 error",
  ],
]
`;

exports[`with comments > with 1 each > cp.spawnSync 1`] = `
[
  [
    "/path/to/node",
    [
      "/path/to/pyright/dist/index.js",
      "--outputjson",
    ],
    {
      "encoding": "utf8",
      "maxBuffer": 104857600,
      "stdio": [
        "ignore",
        "pipe",
        "inherit",
      ],
    },
  ],
]
`;

exports[`with comments > with 1 each > process.chdir 1`] = `[]`;

exports[`with comments > with diagnostics > command.issueCommand 1`] = `
[
  [
    "error",
    {
      "col": 1,
      "file": "/path/to/file1.py",
      "line": 1,
    },
    "some error",
  ],
  [
    "warning",
    {
      "col": 1,
      "file": "/path/to/file2.py",
      "line": 1,
    },
    "some warning",
  ],
  [
    "warning",
    {
      "col": 1,
      "file": "/path/to/file3.py",
      "line": 1,
    },
    "another warning (reportSomeWarning)",
  ],
  [
    "error",
    {
      "col": 10,
      "file": "/path/to/file1.py",
      "line": 6,
    },
    "some error",
  ],
]
`;

exports[`with comments > with diagnostics > core.info 1`] = `
[
  [
    "pyright 1.1.240, node v20.8.1, pyright-action 1.1.0",
  ],
  [
    "Working directory: /some/wd",
  ],
  [
    "Running: /path/to/node /path/to/pyright/dist/index.js --outputjson",
  ],
  [
    "/path/to/file1.py:1:1 - error: some error",
  ],
  [
    "/path/to/file2.py: warning: some warning",
  ],
  [
    "/path/to/file3.py:1:1 - information: some info (reportSomeInformation)",
  ],
  [
    "/path/to/file3.py: warning: another warning (reportSomeWarning)",
  ],
  [
    "/path/to/file1.py:6:10 - error: some error",
  ],
  [
    "2 errors, 2 warnings, 1 information",
  ],
]
`;

exports[`with comments > with diagnostics > core.setFailed 1`] = `
[
  [
    "2 errors",
  ],
]
`;

exports[`with comments > with diagnostics > cp.spawnSync 1`] = `
[
  [
    "/path/to/node",
    [
      "/path/to/pyright/dist/index.js",
      "--outputjson",
    ],
    {
      "encoding": "utf8",
      "maxBuffer": 104857600,
      "stdio": [
        "ignore",
        "pipe",
        "inherit",
      ],
    },
  ],
]
`;

exports[`with comments > with diagnostics > process.chdir 1`] = `[]`;
